src/itkScancoImageIO.cxx

```diff
@@ -1013,16 +1013,27 @@ ScancoImageIO::ReadImageInformation()
   infile.close();
 
   // This code causes rescaling to Hounsfield units
-  /*
   if (this->m_MuScaling > 0 && this->m_MuWater > 0)
     {
-    // HU = 1000*(u - u_water)/u_water
-    this->m_RescaleSlope = 1000.0/(this->m_MuWater * this->m_MuScaling);
-    this->m_RescaleIntercept = -1000.0;
+      // mu(voxel) = intensity(voxel) / m_MuScaling 
+      // HU(voxel) = mu(voxel) * 1000/m_MuWater - 1000
+      // Or, HU(voxel) = intensity(voxel) * (1000 / m_MuWater * m_MuScaling) - 1000
+      this->m_RescaleSlope = 1000.0/(this->m_MuWater * this->m_MuScaling);
+      this->m_RescaleIntercept = -1000.0;
     }
-  */
 }
 
+template <typename TBufferType>
+void 
+RescaleToHU(TBufferType * buffer, size_t size, double slope, double intercept)
+{
+  for (size_t i = 0; i < size; i++)
+  {
+    double bufferValue = static_cast<double>(buffer[i]);
+    bufferValue = bufferValue * slope + intercept;
```

Comment:
Are you forcing this computation into double type (by cast on the line before) due to fear of losing precision? This soft of operation has great potential to be vectorized, and float run about twice as fast as doubles in that case.

Reply:
That was the idea, but I can change the doubles to floats.